relayer/app/state.go

```diff
@@ -0,0 +1,90 @@
+package relayer
+
+import (
+	"encoding/json"
+	"os"
+	"sync"
+
+	"github.com/omni-network/omni/lib/errors"
+
+	"github.com/cometbft/cometbft/libs/tempfile"
+)
+
+type State interface {
```

Comment:
no need to interface if you are not mocking it. Using real implementation is ok in tests. So interface not needed.

Reply:
Maybe rename to `Cursors`